### PR TITLE
Various optimizations

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
`paket install` on a solution containing about 200 projects, 100 packages:

before: 17 seconds
after: 11 seconds.

### CODE REVIEW

**Comment by forki** [comment]
oh

**Comment by vasily-kirichenko** [comment]
;)

File: src/Paket.Core/Utils.fs [hunk 1]
```diff
@@ -19,7 +19,17 @@ let notNullOrEmpty = not << System.String.IsNullOrEmpty
 
 let inline force (lz: 'a Lazy)  = lz.Force()
 let inline endsWith text x = (^a:(member EndsWith:string->bool)x, text) 
-let inline toLower str = (^a:(member ToLower:unit->string)str) 
+let inline toLower str = (^a:(member ToLower:unit->string)str)
+
+let memoze (f: 'a -> 'b) : 'a -> 'b =
```
Comment (adamchester) [inline src/Paket.Core/Utils.fs]: Should this be named `memoize`?

**Comment by vasily-kirichenko** [comment]
@adamchester done

**Pull request merged**